rotkehlchen/chain/ethereum/tokens.py

```diff
@@ -84,46 +102,58 @@ def _get_token_balances(
         """
         log.debug(
             'Querying ethereum chain for multi token account balances',
-            eth_address=account,
+            eth_addresses=accounts,
             tokens_num=len(tokens),
         )
         result = ETH_SCAN.call(
             ethereum=self.ethereum,
-            method_name='tokensBalance',
-            arguments=[account, [x.ethereum_address for x in tokens]],
+            method_name='tokensBalances',
+            arguments=[accounts, [token.ethereum_address for token in tokens]],
             call_order=call_order,
         )
-        balances: Dict[EthereumToken, FVal] = defaultdict(FVal)
-        for token_balance, token in zip(result, tokens):
-            if token_balance == 0:
-                continue
-
-            normalized_balance = token_normalized_value(token_balance, token)
-            log.debug(
-                f'Found {token.symbol}({token.ethereum_address}) token balance for '
-                f'{account} and balance {normalized_balance}',
-            )
-            balances[token] += normalized_balance
+        balances: TokenBalancesType = defaultdict(lambda: defaultdict(FVal))
+        for account, tokens_balances in zip(accounts, result):
+            for token, token_balance in zip(tokens, tokens_balances):
+                if token_balance == 0:
+                    continue
+
+                normalized_balance = token_normalized_value(token_balance, token)
+                log.debug(
+                    f'Found {token.symbol}({token.ethereum_address}) token balance for '
+                    f'{account} and balance {normalized_balance}',
+                )
+                balances[account][token] += normalized_balance
         return balances
 
-    def _query_chunks(
+    def _chunk_and_query(
             self,
-            account: ChecksumEthAddress,
-            chunks: Iterable[List[EthereumToken]],
-            querying_etherscan: bool,
-    ) -> Dict[EthereumToken, FVal]:
-        total_token_balances: Dict[EthereumToken, FVal] = defaultdict(FVal)
-        for chunk in chunks:
-            if querying_etherscan is True:
-                call_order = [ETHERSCAN_NODE]
-            else:
-                call_order = self.ethereum.default_call_order()
-            new_token_balances = self._get_token_balances(
-                account=account,
-                tokens=chunk,
-                call_order=call_order,
-            )
-            total_token_balances = combine_dicts(total_token_balances, new_token_balances)
+            accounts: List[ChecksumEthAddress],
+            tokens: List[EthereumToken],
+    ) -> TokenBalancesType:
+        if self.ethereum.connected_to_any_web3():
+            querying_etherscan = False
+            chunk_size = OTHER_MAX_TOKEN_CHUNK_LENGTH
+        else:
+            querying_etherscan = True
+            chunk_size = ETHERSCAN_MAX_TOKEN_CHUNK_LENGTH
```

Comment:
Why not just
```suggestion
        if self.ethereum.connected_to_any_web3():
            call_order = self.ethereum.default_call_order()
            chunk_size = OTHER_MAX_TOKEN_CHUNK_LENGTH
        else:
            call_order = [ETHERSCAN_NODE]
            chunk_size = ETHERSCAN_MAX_TOKEN_CHUNK_LENGTH
```

Reply:
Ah yeah, sure. `querying_etherscan` boolean was needed before but now it surely can be removed

Reply:
The reason is that each time we call `default_call_order`, a new order of node is generated. So I thought it would make sense to get new call sequence each time we query tokens (to not overload any of the nodes)

Reply:
it seems you did this change so will mark convo as resolved